### PR TITLE
Swagger preview version(2020-09-01-preview) for Microsoft.DataBoxEdge

### PR DESCRIPTION
**Pull request by pvsraviteja**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json [hunk 1]
```diff
@@ -756,12 +756,58 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/generateCertificate": {
```
Comment (weidongxu-microsoft) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:759]: Are these new API, or API already existed but not documented before?

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json [hunk 2]
```diff
@@ -756,12 +756,58 @@
...
+      "post": {
+        "tags": [
+          "Devices"
+        ],
+        "description": "Generates certificate for activation key.",
+        "operationId": "Devices_GenerateCertificate",
+        "parameters": [
+          {
+            "name": "deviceName",
```
Comment (weidongxu-microsoft) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:768]: Suggestion is to put `deviceName` after `resourceGroupName`. But if most API in PR is written this way, no need to change.

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json (hunk 1 above)
Comment (pvsraviteja) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:759]: This is a new API which is added.

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json (hunk 2 above)
Comment (pvsraviteja) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:768]: ok thanks

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json [hunk 3]
```diff
@@ -0,0 +1,8511 @@
...
+            "description": "Successfully deleted the addon."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
```
Comment (j5lim) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:1804]: Doesn't the API return 'Location' header to track async DELETE?

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json [hunk 4]
```diff
@@ -0,0 +1,8511 @@
...
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "name": "$expand",
+            "in": "query",
+            "description": "Specify $filter='CustomContextTag eq <tag>' to filter on custom context tag property",
```
Comment (j5lim) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:3191]: The description is not for $expand.

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json [hunk 5]
```diff
@@ -0,0 +1,8511 @@
...
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "name": "$expand",
+            "in": "query",
+            "description": "Specify $filter='Type eq <type>' to filter on user type property",
```
Comment (j5lim) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:3575]: Update the description.

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/examples/RoleListAddOns.json [hunk 6]
```diff
@@ -0,0 +1,64 @@
+{
+  "parameters": {
+    "api-version": "2020-09-01-preview",
+    "subscriptionId": "4385cf00-2d3a-425a-832f-f4285b1c9dce",
+    "resourceGroupName": "GroupForEdgeAutomation",
+    "deviceName": "testedgedevice",
+    "roleName": "IoTRole1",
+    "accept-language": [
+      "en-US"
+    ]
```
Comment (j5lim) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/examples/RoleListAddOns.json:10]: The parameters do not match the example responses.

**Review by j5lim — CHANGES_REQUESTED** [review]

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json [hunk 7]
```diff
@@ -0,0 +1,8511 @@
...
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "DownloadUpdatesPost": {
```
Comment (mentat9) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:799]: Example should be for GenerateCertificate

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json [hunk 8]
```diff
@@ -0,0 +1,8511 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/getExtendedInformation": {
+      "post": {
+        "tags": [
+          "Devices"
+        ],
+        "description": "Gets additional information for the specified Azure Stack Edge/Data Box Gateway device.",
```
Comment (mentat9) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:810]: Is Data Box Edge renamed to Azure Stack Edge?

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json [hunk 9]
```diff
@@ -0,0 +1,8511 @@
...
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The list of Nodes.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Node"
+          },
+          "readOnly": true
```
Comment (mentat9) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:6738]: Consider making this list pageable and adding a next link property

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json [hunk 10]
```diff
@@ -0,0 +1,8511 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataBoxEdge/dataBoxEdgeDevices/{deviceName}/updateExtendedInformation": {
+      "post": {
+        "tags": [
+          "Devices"
+        ],
+        "description": "Gets additional information for the specified Data Box Edge/Data Box Gateway device.",
```
Comment (mentat9) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:3397]: Action name in URL is "update", not "get". Also, is there a way to GET this info?

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json [hunk 11]
```diff
@@ -0,0 +1,8511 @@
...
+      "description": "Represent the secrets intended for encryption with asymmetric key pair.",
+      "required": [
+        "value",
+        "encryptionAlgorithm"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The value of the secret.",
+          "type": "string"
```
Comment (mentat9) [inline specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:4128]: This should be marked as x-ms-secret and nulled out when present in GET.

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json (hunk 8 above)
Comment (pvsraviteja) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:810]: yes

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json [hunk 12]
```diff
@@ -0,0 +1,8511 @@
...
+    },
+    "GenerateCertResponse": {
+      "description": "Used in activation key generation flow.",
+      "type": "object",
+      "properties": {
+        "publicKey": {
+          "description": "Gets or sets base64 encoded certificate raw data,\r\nthis is the public part needed to be uploaded to cert vault",
+          "type": "string"
+        },
+        "privateKey": {
```
[the remark this answers is not in the document]
Comment (pvsraviteja) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:5284]: Added

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json [hunk 13]
```diff
@@ -0,0 +1,8511 @@
...
+          "description": "DCAccessCode properties.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "DCAccessCodeProperties": {
+      "description": "DCAccessCode Properties.",
+      "type": "object",
+      "properties": {
+        "authCode": {
```
[the remark this answers is not in the document]
Comment (pvsraviteja) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:5114]: Not a secret.

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json (hunk 3 above)
Comment (pvsraviteja) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:1807]: It returns azure async repsonse header which is tracked.

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json (hunk 4 above)
Comment (pvsraviteja) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:3194]: Changed

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json (hunk 5 above)
Comment (pvsraviteja) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json]: Changed

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/examples/RoleListAddOns.json (hunk 6 above)
Comment (pvsraviteja) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/examples/RoleListAddOns.json:10]: Changed

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json (hunk 7 above)
Comment (pvsraviteja) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json]: Changed

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json (hunk 9 above)
Comment (pvsraviteja) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:6743]: Changed

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json (hunk 10 above)
Comment (pvsraviteja) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:3400]: There is only one API which is GetUpdateExtendedInfo. This is existing API and is not added in this iteration.

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json (hunk 11 above)
Comment (pvsraviteja) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:4132]: Added.

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json (hunk 3 above)
Comment (j5lim) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:1807]: Then please add `"x-ms-long-running-operation-options" : { "final-state-via" : "location" }` By default it assumes using Azure-AsyncOperation header.
https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation

**Review by j5lim — APPROVED** [review]
Signing from ARM.

File: specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json (hunk 3 above)
Comment (pvsraviteja) [reply specification/databoxedge/resource-manager/Microsoft.DataBoxEdge/preview/2020-09-01-preview/databoxedge.json:1807]: Changed as requested.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**